_posts/2016-11-29-future.md

```diff
@@ -0,0 +1,269 @@
+
+> There are a few plugins that actually do type checking: [babel-plugin-typecheck](https://github.com/codemix/babel-plugin-typecheck) and [babel-plugin-tcomb](https://github.com/gcanti/babel-plugin-tcomb)
+
+### Take in a Dependency Graph / Operate Multifile
+
+Then we could integrate with other tools like Webpack better. This would allow cross file transformations or whole codebase optimizations. The main reason for this would be for the minifier (being able to remove properties based on checking the usage across the whole application) or for example providing errors for missing/invalid import/exports.
+
+### Parser Errors
+
+Better error messages from the parser like in [Compiler Errors for Humans](http://elm-lang.org/blog/compiler-errors-    for-humans).
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">babel-eslint@7.1.1: now adds the code frame when there&#39;s a parser error! <a href="https://t.co/yoxRpGXq5E">pic.twitter.com/yoxRpGXq5E</a></p>&mdash; Henry Zhu (@left_pad) <a href="https://twitter.com/left_pad/status/799388723896946692">November 17, 2016</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+> It is obvious that we all want to see helpful errors!
+
+- [babel/babylon#125](https://github.com/babel/babylon/pull/125): Instead of "Unexpected token" explain that `await` needs to be used in an async function.
+- Do better inference/guessing on user intention to prevent vague errors.
+
+### `babel-init`
+
+Basically a way to setup Babel easier like create-react-app does.
+
+- Setup a `.babelrc` from scratch
+- Maybe reuse the `babel` npm package to default to use babel-preset-env
+
+### Run [tc39/test262](https://github.com/tc39/test262)
+
+Running official spec tests against Babel to make sure we comply to the spec or at least know when we don't.
+
+### Changing the Mindset Around Stage X Presets
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">My rule of thumb on how I decide what future features to transpile:<br>&quot;Could I reasonably codemod this if it changes?&quot;<br>Don&#39;t do it otherwise.</p>&mdash; Kent C. Dodds (@kentcdodds) <a href="https://twitter.com/kentcdodds/status/803815749416456196">November 30, 2016</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+Maybe we should rename or alias the stage x presets to explain more that they are "experimental", "dangerous", or likely to change?
+
+### Smoke Tests
+
+It would be useful to either have a reverse [Greenkeeper](https://greenkeeper.io) or to run tests with master branch of Babel so that we can catch any major regressions before any release (node has the [citgm project](https://github.com/nodejs/citgm) for this). In theory we would want to take the biggest open source projects that use Babel and run their tests.
```

Comment:
I know the folks at GreenKeeper have talked about adding this feature; and were showcasing some of their future plans at a conference I was at this year. I'd love this functionality for yargs/nyc.